src/test/java/io/cdap/plugin/MultiFieldAdderConfigTest.java

```diff
@@ -0,0 +1,101 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin;
+
+import io.cdap.cdap.etl.api.validation.CauseAttributes;
+import io.cdap.cdap.etl.api.validation.ValidationFailure;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Unit tests for MultiFieldAdderConfig.
+ */
+public class MultiFieldAdderConfigTest {
+
+  private static final String MOCK_STAGE = "mockStage";
+  private static final MultiFieldAdderConfig VALID_CONFIG = new MultiFieldAdderConfig(
+    "test_1:s_value_1,test_2:s_value_2,test_3:s_value_3"
+  );
+
+  @Test
+  public void testValidConfig() {
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    VALID_CONFIG.validate(failureCollector);
+    Assert.assertTrue(failureCollector.getValidationFailures().isEmpty());
+  }
+
+  @Test
+  public void testValidateDuplicateFieldName() {
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    MultiFieldAdderConfig config = MultiFieldAdderConfig.builder(VALID_CONFIG)
+      .setFieldValue("test_1:s_value_1,test_1:s_value_2")
+      .build();
+    List<List<String>> paramName = Collections.singletonList(
+      Collections.singletonList(MultiFieldAdderConfig.FIELD_VALUE));
+
+    config.validate(failureCollector);
+    assertValidationFailed(failureCollector, paramName);
+  }
+
+  @Test
+  public void testValidateFieldNameWithoutValue() {
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    MultiFieldAdderConfig config = MultiFieldAdderConfig.builder(VALID_CONFIG)
+      .setFieldValue("test_1:s_value_1,test_2,test_3:s_value_3")
+      .build();
+
+    config.validate(failureCollector);
+    Assert.assertTrue(failureCollector.getValidationFailures().isEmpty());
+  }
+
+  @Test
+  public void testValidateDuplicateFieldNameWithoutValue() {
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    MultiFieldAdderConfig config = MultiFieldAdderConfig.builder(VALID_CONFIG)
+      .setFieldValue("test_1:s_value_1,test_2,test_2:s_value_2")
+      .build();
+
+    config.validate(failureCollector);
+    Assert.assertTrue(failureCollector.getValidationFailures().isEmpty());
+  }
+
+  private static void assertValidationFailed(MockFailureCollector failureCollector, List<List<String>> paramNames) {
```

Comment:
move all validation helpers to utility class. See https://github.com/data-integrations/whole-file-ingest/pull/7/files#diff-a222ac89d65b50efb72de4d23fb0ae28

Reply:
Done.